src/ai/ai.cpp

```diff
@@ -2758,56 +2758,69 @@ bool will_accept_peace_offer_value(sys::state& state,
 	int32_t war_duration, bool contains_sq) {
 	bool is_attacking = !offer_from_attacker;
 
-	auto overall_score = primary_warscore;
+	auto overall_score = military::primary_warscore(state, w);
 	if(!is_attacking)
 		overall_score = -overall_score;
 
-	int32_t personal_po_value = target_personal_po_value;
-
 	if(concession && overall_score <= -50.0f) {
 		return true;
 	}
+
 	if(!concession) {
 		overall_po_value = -overall_po_value;
 	}
+	if(overall_po_value < -100)
+		return false;
 
-	auto personal_score_saved = personal_po_value - potential_peace_score_against;
+	auto personal_score_saved = target_personal_po_value - potential_peace_score_against;
 
 	if((prime_attacker == n || prime_defender == n) && (prime_attacker == from || prime_defender == from)) {
 		if(overall_score <= -50 && overall_score <= overall_po_value * 2)
 			return true;
 
-		if(concession && my_side_peace_cost <= overall_po_value)
+		auto war_duration = state.current_date.value - state.world.war_get_start_date(w).value;
+		if(concession && (is_attacking ? military::attacker_peace_cost(state, w) : military::defender_peace_cost(state, w)) <= overall_po_value)
 			return true; // offer contains everything
-
 		if(war_duration < 365) {
 			return false;
 		}
 		float willingness_factor = float(war_duration - 365) * 10.f / 365.0f;
 		if(overall_score >= 0) {
-			if(concession && (overall_score * 2 - overall_po_value - willingness_factor) < 0)
+			if(concession && ((overall_score * 2 - overall_po_value - willingness_factor) < 0))
 				return true;
 		} else {
-			if(overall_score <= overall_po_value && (overall_score / 2 - overall_po_value - willingness_factor) < 0)
+			if((overall_score - willingness_factor) <= overall_po_value && (overall_score / 2 - overall_po_value - willingness_factor) < 0)
```

Comment:
do these changes match the function below?

Reply:
yes